.github/workflows/build_release.yml

```diff
@@ -87,3 +87,59 @@ jobs:
         file: ./${{ steps.archive.outputs.shasum }}
         asset_name: ${{ steps.archive.outputs.shasum }}
         tag: ${{ github.ref }}
+
+  build-wasi:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        submodules: true
+
+    - name: built-tools
+      run: |
+        docker run -di --name wasi-sdk -v $(pwd):/src --workdir /src ghcr.io/webassembly/wasi-sdk:wasi-sdk-20 bash
+        docker exec wasi-sdk cmake -S . -B out -G Ninja -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTS=0
+        # We only build tools, not runtime, for WASI build
+        docker exec wasi-sdk cmake --build out --config Release \
+        --target wast2json-copy-to-bin \
+        --target wasm-decompile-copy-to-bin \
+        --target wasm-opcodecnt-copy-to-bin \
+        --target wasm-objdump-copy-to-bin \
+        --target wasm-interp-copy-to-bin \
+        --target wasm2wat-copy-to-bin \
+        --target wasm-strip-copy-to-bin \
+        --target wat-desugar-copy-to-bin \
+        --target wasm-validate-copy-to-bin \
+        --target spectest-interp-copy-to-bin \
+        --target wat2wasm-copy-to-bin \
+        --target wasm2c-copy-to-bin
+
+    - name: archive
+      id: archive
+      run: |
+        OSNAME=wasi
+        VERSION=${{ github.event.release.tag_name }}
+        PKGNAME="wabt-$VERSION-$OSNAME"
+        TARBALL=$PKGNAME.tar.gz
+        SHASUM=$PKGNAME.tar.gz.sha256
+        mv bin wabt-$VERSION
+        tar -czf $TARBALL wabt-$VERSION
+        scripts/sha256sum.py $TARBALL > $SHASUM
+        echo "::set-output name=tarball::$TARBALL"
+        echo "::set-output name=shasum::$SHASUM"
+
+    - name: upload tarball
+      uses: svenstaro/upload-release-action@v2
+      with:
+        repo_token: ${{ secrets.GITHUB_TOKEN }}
+        file: ./${{ steps.archive.outputs.tarball }}
+        asset_name: ${{ steps.archive.outputs.tarball }}
+        tag: ${{ github.ref }}
+
+    - name: upload shasum
+      uses: svenstaro/upload-release-action@v2
+      with:
+        repo_token: ${{ secrets.GITHUB_TOKEN }}
+        file: ./${{ steps.archive.outputs.shasum }}
+        asset_name: ${{ steps.archive.outputs.shasum }}
+        tag: ${{ github.ref }}
```

Comment:
Can we avoid duplicated these last 3 jobs?  Perhaps we can factor it out into a named job that can be called from both places (I assume github actions supports that).

Reply:
Sure, extracted a reusable action

Reply:
Great!